### PR TITLE
[New features] Support Identity Hyper-Connections (iHC) module

### PR DESCRIPTION
**Pull request by cuichenxu**
## Summary

Add Identity Hyper-Connections (iHC) support. iHC fixes H_res as the identity matrix and drops the learned H_res mapping (no Sinkhorn projection, no n*n mapping head); H_pre and H_post remain fully dynamic and learnable.

## Changes

- **hyper_connection.py**: new `IdentityHyperConnectionModule` (alias iHC), same interface as `HyperConnectionModule` (mHC)
- **transformer_config.py**: new `enable_identity_hyper_connections` flag
- **gpt_layer_specs.py**: wire iHC into `get_gpt_layer_local_spec`, mHC remains the default

## Test

- `mhc_single_stream_init` config field already exists upstream (reused)
- All module interfaces mirror HyperConnectionModule, mHC path untouched

### CODE REVIEW

File: src/paddlefleet/transformer/transformer_config.py [hunk 1]
```diff
@@ -940,6 +940,16 @@ class TransformerConfig(ModelParallelConfig):
     uniform doubly-stochastic matrix: every sub-layer reads and writes an
     averaged mixture of the n residual streams from step 0."""
 
+    enable_identity_hyper_connections: bool = False
```
Comment (risemeup1111) [inline src/paddlefleet/transformer/transformer_config.py:943]: P1 `TransformerConfig.from_config` 使用 `object.__new__`，只注册输入对象中实际存在的键；旧版启用 HC 的配置通常没有这个新字段，因此实例不会获得 `enable_identity_hyper_connections`。随后 `get_gpt_layer_local_spec` 直接读取该属性会在构建模型时触发 `AttributeError`，破坏已有 mHC 配置的向后兼容。请在 `from_config` 补齐 dataclass 默认值，或在消费点使用默认值，并加入旧配置回归测试。

File: src/paddlefleet/models/gpt/gpt_layer_specs.py [hunk 2]
```diff
@@ -664,16 +664,23 @@ def get_gpt_layer_local_spec(
...
     mlp_hc_spec = IdentityOp
     if config is not None and config.enable_hyper_connections:
         from paddlefleet.transformer.hyper_connection import (
             HyperConnectionModule,
+            IdentityHyperConnectionModule,
         )
 
-        self_attention_hc_spec = LayerSpec(layer=HyperConnectionModule)
-        mlp_hc_spec = LayerSpec(layer=HyperConnectionModule)
+        if config.enable_identity_hyper_connections:
```
Comment (risemeup1111) [inline src/paddlefleet/models/gpt/gpt_layer_specs.py:676]: P1 这里只有在 `enable_hyper_connections` 已开启时才读取 iHC 开关；用户单独设置 `enable_identity_hyper_connections=True` 时，GPT/MTP 会静默构建普通 `IdentityOp`/非 HC 路径，配置名所承诺的 iHC 不会生效。请在 `TransformerConfig.__post_init__` 对依赖关系无条件报错，或让 iHC 开关明确联动基础 HC，并覆盖该组合测试。

File: src/paddlefleet/transformer/hyper_connection.py [hunk 3]
```diff
@@ -1068,3 +1069,497 @@ def forward(self, dict_args: dict) -> dict:
...
+        # H_res is fixed identity: no learned projection needed
+        self.mapping_proj = nn.Linear(
+            self.n * self.hidden_size,
+            2 * self.n,
+            bias_attr=False,
+        )
+
+        init_alpha = config.mhc_init_gating_factor
+        # Learnable scaling factors for H_pre and H_post only
+        self.alpha_pre = self.create_parameter(
```
Comment (risemeup1111) [inline src/paddlefleet/transformer/hyper_connection.py:1114]: P1 现有 `HyperConnectionModule` 会把 mapping projection、alpha 和 bias 全部固定为 FP32，并通过 `_cast_to_low_precision=False` 保证 `high_precision_mhc` 在 BF16 AMP 下成立；这里却让 `alpha_pre/alpha_post` 使用 `config.params_dtype`，mapping/bias 也没有同样的 FP32 与 AMP 保留设置。BF16 配置下 iHC 的门控参数会被低精度存储/转换，`high_precision_mhc=True` 仍无法提供与 mHC 一致的高精度映射。请复用 mHC 的 FP32 参数初始化和 AMP 保留标记，并补充 BF16 前向/反向验证。

File: src/paddlefleet/transformer/hyper_connection.py [hunk 4]
```diff
@@ -1068,3 +1069,497 @@ def forward(self, dict_args: dict) -> dict:
                 )
             )
         return dict_args
+
+class IdentityHyperConnectionModule(nn.Layer):
```
Comment (risemeup1111) [inline src/paddlefleet/transformer/hyper_connection.py:1073]: P1 本 PR 新增约 495 行 iHC 前向、初始化、BDA 和配置路由，但 diff 没有任何测试变更；现有用例只覆盖 `HyperConnectionModule`，无法验证 iHC 的 identity `h_res`、`mhc_single_stream_init`、BF16/AMP、bias/dropout 及 GPT/MTP 集成。请在现有 hyper-connection 单卡测试中补齐默认/非默认配置、形状与数值/梯度断言，并加入层 spec 路由回归，必要时覆盖多卡初始化。

**Review by risemeup1111 — CHANGES_REQUESTED** [review]
### Paddle-Bot Review Board （review完成)
| 序号 | 位置 | 优先级 | 规则来源 | 状态 |
| --- | --- | --- | --- | --- |
| 1 | [旧配置字段缺失](https://github.com/PaddlePaddle/PaddleFleet/pull/1894#discussion_r3877789544) | P1 | [仓库规则：用户配置开关](https://github.com/PaddlePaddle/PaddleFleet/blob/d6610a5d1c175690409c49b0b6e48450577a6935/.agents/skills/ai-review/references/user-configuration-rules.md) | 🚧 |
| 2 | [iHC开关依赖](https://github.com/PaddlePaddle/PaddleFleet/pull/1894#discussion_r3877791798) | P1 | [仓库规则：用户配置开关](https://github.com/PaddlePaddle/PaddleFleet/blob/d6610a5d1c175690409c49b0b6e48450577a6935/.agents/skills/ai-review/references/user-configuration-rules.md) | 🚧 |
| 3 | [门控参数精度](https://github.com/PaddlePaddle/PaddleFleet/pull/1894#discussion_r3877793923) | P1 | [仓库规则：基础正确性](https://github.com/PaddlePaddle/PaddleFleet/blob/d6610a5d1c175690409c49b0b6e48450577a6935/.agents/skills/ai-review/references/base-rules.md) | 🚧 |
| 4 | [iHC测试覆盖](https://github.com/PaddlePaddle/PaddleFleet/pull/1894#discussion_r3877795595) | P1 | [仓库规则：用户配置开关](https://github.com/PaddlePaddle/PaddleFleet/blob/d6610a5d1c175690409c49b0b6e48450577a6935/.agents/skills/ai-review/references/user-configuration-rules.md)<br>[仓库规则：测试质量](https://github.com/PaddlePaddle/PaddleFleet/blob/d6610a5d1c175690409c49b0b6e48450577a6935/.agents/skills/ai-review/references/base-rules.md) | 🚧 |

<div align="right">
   <sup>Powered by Nyanpasu with gpt-5.6-sol 默认推理级别, please check the suggestions carefully.</sup>
</div>

File: src/paddlefleet/transformer/hyper_connection.py (hunk 3 above)
Comment (risemeup1111) [reply src/paddlefleet/transformer/hyper_connection.py:1114]: 新增注释说明了 `alpha_res` 只是 AOA 的占位参数，但没有改变精度行为：iHC 的 `mapping_proj`、`alpha_*` 和 `bias` 仍未像 mHC 一样固定为 FP32 并设置 AMP 保留标记。因此该 P1 仍未解决，请按原建议修复并补充 BF16 前向/反向验证。

File: src/paddlefleet/transformer/hyper_connection.py (hunk 4 above)
Comment (risemeup1111) [reply src/paddlefleet/transformer/hyper_connection.py:1073]: 当前提交仅修改 `alpha_res` 注释，仍没有新增 iHC 或路由测试；该 P1 仍未解决。

**Review by risemeup1111 — CHANGES_REQUESTED** [review]
